src/Text.mo

```diff
@@ -1,23 +1,42 @@
 /// Text values
 ///
-/// This type describes a valid, human-readable text. It does not contain arbitrary
-/// binary data.
+/// This type represents human-readable text as sequences of characters of type [`Char`](Char.html) .
+/// If `t` is a value of type `Text`, then:
+/// * `t.chars()` returns an _iterator_ of type `Iter<Char>` enumerating its characters from first to last.
+/// * `t.size()` returns the _size_ (or length) of `t` (and `t.chars()`) as a `Nat`.
+///
+/// This module defines additional operations on `Text` values.
 
+import Char "Char";
 import Iter "Iter";
 import Hash "Hash";
 import Prim "mo:prim";
 
 module {
 
-  /// Returns the concatenation of `x` and `y`, `x # y`.
-  public func concat(x : Text, y : Text) : Text =
-    x # y;
+  /// Conversion.
+  /// Returns the text value of size 1 containing the single character `c`.
+  public let fromChar : (c : Char) -> Text = Prim.charToText;
+
+  /// Conversion.
+  /// Creates an [iterator](Iter.html#type.Iter) that traverses the characters of the text `t`.
+  public func toIter(t : Text) : Iter.Iter<Char> =
+    t.chars();
+
+  /// Conversion.
+  /// Returns the text value containing the sequence of characters in `cs`.
+  public func fromIter(cs : Iter.Iter<Char>) : Text {
+    var r = "";
+    for (c in cs) {
+      r #= Prim.charToText(c);
+    };
+    return r;
+  };
 
-  /// Creates an [iterator](Iter.html#type.Iter) that traverses the characters of the text.
-  public func toIter(text : Text) : Iter.Iter<Char> =
-    text.chars();
+  /// Returns `t.size()`, the number of characters in `t` (and `t.chars()`).
+  public func size(t : Text) : Nat { t.size(); };
 
-  /// WARNING: This only hashes the lowest 32 bits of the `Int`
+  /// Returns a hash obtained by the `xor`-ing the (`Word32`) values of all characters in `t`.
```

Comment:
We should do something else, I think, even if it's just the [Jenkin's hash function](https://en.wikipedia.org/wiki/Jenkins_hash_function), already in `Hash` module.

Reply:
https://github.com/dfinity/motoko-base/blob/d0354489313e7334054505abe04574482d409cbd/src/Hash.mo#L54

Reply:
I'm actual wondering where these gash functions should live. Only some types have them now, which is odd. What should we use for words etc, just the underlying bits. Do we care about collision at different types?